### PR TITLE
KeymanWeb - Unit Testing:  Control Attachment

### PR DESCRIPTION
**Pull request by jahorton**
While not yet a fully robust test, the newly-introduced unit test file begins the conversion of the manual `testing/attachment-api/` testing page into an automated unit test.

At present, this serves to ensure that control attachment behaves exactly as expected for both the 'desktop' and 'touch' usage paradigms.

It also expands the testing infrastructure somewhat, adding a unit_tests/resources section that can be imported and linked to by unit tests without fear of pre-processing.

Finally, it fixes the `fsevents` npm warnings that were happening every build.

### CODE REVIEW

File: web/unit_tests/cases/attachmentAPI.js [hunk 1]
```diff
@@ -0,0 +1,215 @@
+var assert = chai.assert;
+
+var DynamicElements;
+var inputCounter = 0;
+
+if(typeof(DynamicElements) == 'undefined') {
+  DynamicElements = {};
+
+  DynamicElements.addInput = function() {
+	  var masterDiv = document.getElementById('DynamicElements');
```
Comment (mcdurdin) [inline web/unit_tests/cases/attachmentAPI.js]: indenting looks off in this file; can you convert to spaces (indent = 2)

File: web/unit_tests/cases/attachmentAPI.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+				});
+
+				it('<textarea>', function(done) {
+					var ID = DynamicElements.addText();
+					var ele = document.getElementById(ID);
+
+					DynamicElements.assertAttached(ele, done);
+				});
+
+				// We still have weird collateral issues with IE and Edge if this is enabled.
```
Comment (mcdurdin) [inline web/unit_tests/cases/attachmentAPI.js]: Can you elaborate?

File: web/unit_tests/test_utils.js [hunk 3]
```diff
@@ -30,35 +46,73 @@ var setupKMW = function(ui) {
    */
   window.setTimeout(function() {
     initFunc();
-  }, 5);
+  }, 8);
```
Comment (mcdurdin) [inline web/unit_tests/test_utils.js]: Why the change from 5 to 8? Given `setTimeout` has a minimum timeout of 10, what was the purpose of this?

File: web/unit_tests/test_utils.js [hunk 4]
```diff
@@ -30,35 +46,73 @@ var setupKMW = function(ui) {
...
 
   Lscript.src = src;
 
   return Lscript;
 }
 
 var teardownKMW = function() {
+  keyman['shutdown']();
   delete window.keyman;
   window["keyman"] = undefined;
```
Comment (mcdurdin) [inline web/unit_tests/test_utils.js]: This isn't the best way to remove a property from an object. See https://stackoverflow.com/questions/208105/how-do-i-remove-a-property-from-a-javascript-object

File: web/unit_tests/test_utils.js [hunk 5]
```diff
@@ -30,35 +46,73 @@ var setupKMW = function(ui) {
...
+
+  // We need managed state for this.
+  var InitializationManager = function() {
+    this.killSwitch = false;
+
+    this.initCheckCallback = function() {
+      if(window['keyman'] && window['keyman'].initialized == 2 && uiInitCheck()) {
+        if(done) {
+          this.timer = window.setTimeout(function() { 
+            // There can be some odd cross-interference with the UI modules and their initialization.
```
Comment (mcdurdin) [inline web/unit_tests/test_utils.js]: It'd be good to quantify this. It's all in our grasp; we just need to map it out :).

File: web/unit_tests/cases/attachmentAPI.js (hunk 2 above)
Comment (jahorton) [reply web/unit_tests/cases/attachmentAPI.js]: Leftover comment from doing development; I disabled the following test initially when having those cross-frame script errors that led to the development of the new `keyman.shutdown` function.

File: web/unit_tests/test_utils.js (hunk 3 above)
Comment (jahorton) [reply web/unit_tests/test_utils.js]: I was tweaking that value when figuring out the source of some weird UI interactions when (as it turns out) the post-init timeouts were too low.  I'll put it back.

File: web/unit_tests/test_utils.js (hunk 4 above)
Comment (jahorton) [reply web/unit_tests/test_utils.js]: So... exactly what I did above, but without the `window["keyman"] = undefined` line after it?

File: web/unit_tests/test_utils.js (hunk 5 above)
Comment (jahorton) [reply web/unit_tests/test_utils.js]: It would be nice, yes.  I spent a couple of hours yesterday trying to fine-tune things and find the exact source of the issues, but never quite figured out how the cross-interference was happening.  It may be more trouble than it's worth at the moment; I suppose a new issue for it is in order?

Comment (jahorton) [reply web/unit_tests/test_utils.js]: The most fun instance was where the Float UI was initializing, the window['keyman'] instance of KeymanWeb said the Float UI was initiailizing, but the instance of window['keyman'] the Float UI saw said the Button UI was correct.  So, Float's `initialize()` was being called without its supporting functions... somehow.

File: web/unit_tests/test_utils.js (hunk 4 above)
Comment (mcdurdin) [reply web/unit_tests/test_utils.js]: Correct 👍
